cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -145,10 +148,27 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 	for (size_t i = 0; i < glob_res.gl_pathc; ++i) {
 		char symlink_name[512] = { 0 };
 		char symlink_target[512] = { 0 };
+		char prefix_dir[512] = { 0 };
 		const char *pathname = glob_res.gl_pathv[i];
 		char *pathname_copy
 		    SC_CLEANUP(sc_cleanup_string) = strdup(pathname);
+		char *dirname_copy
+		    SC_CLEANUP(sc_cleanup_string) = strdup(pathname);
 		char *filename = basename(pathname_copy);
+		char *directory_name = dirname(dirname_copy);
+		strncpy(prefix_dir, libgl_dir, sizeof prefix_dir);
```

Comment:
Can you please use `sc_must_snprintf` instead, it's a bit easier to get right than knowing exactly how strncpy fails.